src/main/java/stepsdesign/PipelineSteps.java

```diff
@@ -0,0 +1,134 @@
+  }
+
+  @Then("Verify the preview of pipeline is {string}")
+  public void verifyThePreviewOfPipelineIs(String previewStatus) {
+    WebDriverWait wait = new WebDriverWait(SeleniumDriver.getDriver(), 180);
+    wait.until(ExpectedConditions.visibilityOf(CdfStudioLocators.statusBanner));
+    Assert.assertTrue("Preview of pipeline should " + previewStatus,
+                      CdfStudioLocators.statusBannerText.getText().contains(previewStatus));
+    if (!previewStatus.equalsIgnoreCase("failed")) {
+      wait.until(ExpectedConditions.invisibilityOf(CdfStudioLocators.statusBanner));
+    }
+  }
+
+  @Then("Close the preview")
+  public void closeThePreview() {
+    CdfStudioActions.previewSelect();
+  }
+
+  @Then("Close the preview data")
+  public void closeThePreviewData() {
+    CdfStudioActions.clickCloseButton();
+    CdfStudioActions.previewSelect();
+  }
+
+  @Then("Deploy the pipeline")
+  public void deployThePipeline() {
+    SeleniumHelper.waitElementIsVisible(CdfStudioLocators.pipelineDeploy, 2);
+    CdfStudioActions.pipelineDeploy();
+  }
+
+  @Then("Save and Deploy Pipeline")
+  public void saveAndDeployPipeline() {
+    CdfStudioActions.pipelineName();
+    CdfStudioActions.pipelineNameIp("TestPipeline" + UUID.randomUUID());
+    CdfStudioActions.pipelineSave();
+    SeleniumHelper.waitElementIsVisible(CdfStudioLocators.statusBanner);
+    WebDriverWait wait = new WebDriverWait(SeleniumDriver.getDriver(), 5);
+    wait.until(ExpectedConditions.invisibilityOf(CdfStudioLocators.statusBanner));
+    CdfStudioActions.pipelineDeploy();
+  }
+
+  @Then("Run the Pipeline in Runtime")
+  public void runThePipelineInRuntime() throws InterruptedException {
+    CdfPipelineRunAction.runClick();
+  }
+
+  @Then("Wait till pipeline is in running state")
+  public void waitTillPipelineIsInRunningState() {
+    WebDriverWait wait = new WebDriverWait(SeleniumDriver.getDriver(), 300);
+    wait.until(ExpectedConditions.or(
+      ExpectedConditions.visibilityOfElementLocated(By.xpath("//*[@data-cy='Succeeded']")),
+      ExpectedConditions.visibilityOfElementLocated(By.xpath("//*[@data-cy='Failed']"))));
+  }
+
+  @Then("Verify the pipeline status is {string}")
+  public void verifyThePipelineStatusIs(String status) {
+    Assert.assertTrue("Pipeline status should be " + status,
+                      SeleniumHelper.verifyElementPresent("//*[@data-cy='" + status + "']"));
+  }
+
+  @Then("Open and capture logs")
+  public void openAndCaptureLogs() {
+    try {
+      CdfPipelineRunAction.logsClick();
+      BeforeActions.scenario.write(CdfPipelineRunAction.captureRawLogs());
+      PrintWriter out = new PrintWriter(BeforeActions.file);
+      out.println(CdfPipelineRunAction.captureRawLogs());
+    } catch (Exception e) {
+      BeforeActions.scenario.write("Exception in capturing logs : " + e);
+    }
+  }
+
+  @Then("Validate successMessage is displayed")
+  public void validateSuccessMessageIsDisplayed() {
```

Comment:
Please remove this step, don't need this redundant step. `Verify the pipeline status is succeeded` is enough.

Reply:
removed

Reply:
Please squash the commits

Reply:
Done.